src/sources/kafka.rs

```diff
@@ -120,121 +131,92 @@ impl SourceConfig for KafkaSourceConfig {
-
-                            let msg_key = msg
-                                .key()
-                                .map(|key| Value::from(String::from_utf8_lossy(key).to_string()))
-                                .unwrap_or(Value::Null);
-                            log.insert(&key_field, msg_key);
-
-                            log.insert(&topic_key, Value::from(msg.topic().to_string()));
-
-                            log.insert(&partition_key, Value::from(msg.partition()));
-
-                            log.insert(&offset_key, Value::from(msg.offset()));
-
-                            let mut headers_map = BTreeMap::new();
-                            if let Some(headers) = msg.headers() {
-                                // Using index-based for loop because rdkafka's `Headers` trait
-                                // does not provide Iterator-based API
-                                for i in 0..headers.count() {
-                                    if let Some(header) = headers.get(i) {
-                                        headers_map.insert(
-                                            header.0.to_string(),
-                                            Bytes::from(header.1.to_owned()).into(),
-                                        );
-                                    }
-                                }
-                            }
-                            log.insert(headers_key, Value::from(headers_map));
-
-                            consumer.store_offset(&msg).map_err(|error| {
-                                emit!(KafkaOffsetUpdateFailed { error });
-                            })?;
-
-                            Ok(event)
                         }
                     }
                 }
-            })
-            // Try `forward` after removing old futures.
-            // Error: implementation of `futures_core::stream::Stream` is not general enough
-            // .forward(
-            //     out.sink_compat()
-            //         .sink_map_err(|error| error!(message = "Error sending to sink.", %error)),
-            // )
-            .for_each(|item| {
-                let mut out = out.clone();
-                async move {
-                    if let Ok(item) = item {
-                        if let Err(error) = out.send(item).await {
-                            error!(message = "Error sending to sink.", %error);
-                        }
-                    }
+                log.insert(&headers_key, Value::from(headers_map));
+
+                if let Err(error) = consumer.store_offset(&msg) {
+                    emit!(KafkaOffsetUpdateFailed { error });
+                    continue;
                 }
-            })
-            .await;
-        Ok(())
-    }))
+
+                let _ = out
+                    .send(event)
+                    .await
+                    .map_err(|error| error!(message = "Error sending to sink.", %error));
```

Comment:
Also, this might be a little cleaner way to do the same thing:
```suggestion
                if let Some(error) = out.send(event).await {
                    error!(message = "Error sending to sink.", %error);
                }
```